### PR TITLE
Trim build runner apis

### PR DESCRIPTION
**Pull request by jakemac53**
Closes https://github.com/dart-lang/build/issues/1822

Fixes https://github.com/dart-lang/build/issues/1188

Will wait to merge until 0.10.3 is published

### CODE REVIEW

File: _test/test/goldens/generated_build_script.dart [hunk 1]
```diff
@@ -7,6 +7,7 @@ import 'package:build_web_compilers/builders.dart' as _i6;
 import 'package:build/build.dart' as _i7;
 import 'package:build_vm_compilers/builders.dart' as _i8;
 import 'dart:isolate' as _i9;
+import 'package:build_runner/build_runner.dart' as _i10;
```
Comment (samogot) [inline _test/test/goldens/generated_build_script.dart]: is it possible to move to other package imports?

File: build_runner/CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,21 @@
...
+    build script you will need to import build_runner_core directly and add a
+    dependency on it.
+  - Stopped exporting the `build` and `watch` functions directly, as well as the
+    `ServeHandler`.
+  - If this has broken your use case please file an issue on the package and
+    request that we export the api you were using previously. These will be
+    considered on an individual basis but the bar for additional exports will be
+    high.
+
+## 0.10.3
```
Comment (samogot) [inline build_runner/CHANGELOG.md]: Have we released 0.10.3?

File: _test/test/goldens/generated_build_script.dart (hunk 1 above)
Comment (jakemac53) [reply _test/test/goldens/generated_build_script.dart]: This is used for the `run` method, which is now the only api from the build_runner package

File: build_runner/CHANGELOG.md (hunk 2 above)
Comment (jakemac53) [reply build_runner/CHANGELOG.md]: No - I won't merge this until we have though, just wanted to get the PR out.

**Comment by natebosch** [comment]
cc @kevmoo - I think you're already using only `run` in manual build scripts, but if you have a `watch` or `build` it's going to break.

**Comment by kevmoo** [comment]
I just call `run` the few places I do this...

**Comment by jakemac53** [comment]
Note that you will have to import `build_runner_core` now also, but that should be a trivial change.

**Pull request merged**